scripts/geo2fdn.py

```diff
@@ -0,0 +1,527 @@
+                    fq.write(row, sheet_dict_fq['related_files.relationship_type'], 'paired with')
+                    fq.write(row, sheet_dict_fq['related_files.file'], fq2)
+                    fq.write(row, sheet_dict_fq['read_length'], entry.length)
+                    fq.write(row, sheet_dict_fq['instrument'], entry.instr)
+                    fq.write(row, sheet_dict_fq['dbxrefs'], 'SRA:' + run)
+                    fq.write(row + 1, sheet_dict_fq['aliases'], fq2)
+                    # fq.write(row + 1, sheet_dict_fq['description'], entry.description)
+                    fq.write(row + 1, sheet_dict_fq['*file_format'], 'fastq')
+                    fq.write(row + 1, sheet_dict_fq['paired_end'], '2')
+                    fq.write(row + 1, sheet_dict_fq['related_files.relationship_type'], 'paired with')
+                    fq.write(row + 1, sheet_dict_fq['related_files.file'], fq1)
+                    fq.write(row + 1, sheet_dict_fq['read_length'], entry.length)
+                    fq.write(row + 1, sheet_dict_fq['instrument'], entry.instr)
+                    fq.write(row + 1, sheet_dict_fq['dbxrefs'], 'SRA:' + run)
+                    row += 2
+                elif entry.layout.lower() == 'single':
+                    fq_0 = alias_prefix + ':' + run + '_fq'
+                    file_dict[entry.geo] += [fq_0]
+                    fq.write(row, sheet_dict_fq['aliases'], fq_0)
+                    # fq.write(row, sheet_dict_fq['description'], entry.description)
+                    fq.write(row, sheet_dict_fq['*file_format'], 'fastq')
+                    fq.write(row, sheet_dict_fq['read_length'], entry.length)
+                    fq.write(row, sheet_dict_fq['instrument'], entry.instr)
+                    fq.write(row, sheet_dict_fq['dbxrefs'], 'SRA:' + run)
+                    row += 1
+                else:
+                    raise ValueError("Invalid value for layout. Layout must be 'single' or 'paired'.")
+
+    exp_sheets = [name for name in book.sheet_names() if name.startswith('Experiment')]
+    if len(exp_sheets) > 0:
+
+        hic_expts = [exp for exp in gds.experiments if exp.exptype.startswith('hic') or
+                     exp.exptype.startswith('dnase hic')]
+        if 'ExperimentHiC' in book.sheet_names() and hic_expts:
+            outbook = write_experiments('ExperimentHiC', hic_expts, alias_prefix,
+                                        file_dict, book, outbook)
+        elif 'ExperimentHiC' in book.sheet_names() and not hic_expts:
+            print("\nNo HiC experiments parsed from %s." % geo)
+            print("If all samples are known to be HiC experiments,",
+                  "this script can be rerun using -t HiC")
+        elif 'ExperimentHiC' not in book.sheet_names() and hic_expts:
+            print("\nHiC experiments found in %s but no ExperimentHiC sheet present in workbook." % geo,
+                  "HiC experiments will not be written to file.")
+
```

Comment:
You have a number of repeated patterns starting here dealing with the different experiment types that could probably be handled in a function